TripKitUI/views/TKPathFriendlinessView.swift

```diff
@@ -0,0 +1,132 @@
+  
+  public var segment: TKSegment? {
+    didSet {
+      update()
+    }
+  }
+  
+  public typealias PathFriedliness = (friendly: CGFloat, unfriendly: CGFloat, unknown: CGFloat)
+  
+  public override init(frame: CGRect) {
+    super.init(frame: frame)
+  }
+  
+  public required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+  
+  public override func awakeFromNib() {
+    super.awakeFromNib()
+    
+    friendlyLegendLabel.text = Loc.FriendlyPath
+    unfriendlyLegendLabel.text = Loc.UnfriendlyPath
+    unknownLegendLabel.text = Loc.UnknownPathFriendliness
+  }
+  
+  fileprivate func update() {
+    guard
+      let segment = self.segment,
+      segment.template != nil,
+      let metres = segment.template.metres
+      else { return }
+    
+    let friendlyMetres = segment.template.metresFriendly != nil ? segment.template.metresFriendly.doubleValue : Double(0)
+    let friendlyRatio = friendlyMetres / metres.doubleValue
+    
+    let unfriendlyMetres = segment.template.metresUnfriendly != nil ? segment.template.metresUnfriendly.doubleValue : Double(0)
+    let unfriendlyRatio = unfriendlyMetres / metres.doubleValue
+    
+    let unknownMetres = metres.doubleValue - friendlyMetres - unfriendlyMetres
+    let unknownRatio = unknownMetres / metres.doubleValue
+    
+    let formatter = NumberFormatter()
+    formatter.numberStyle = .percent
+    
+    // Update title
+    let format: String
+    if segment.isCycling() {
+      format = NSLocalizedString("%@ cycle friendly", tableName: "TripKit", bundle: .tripKitUI, comment: "Indicator for how cycle-friendly a cycling route is. Placeholder will get replaced with '75%'.")
+    } else {
+      format = NSLocalizedString("%@ wheelchair friendly", tableName: "TripKit", bundle: .tripKitUI, comment: "Indicator for how wheelchair-friendly a wheeelchair route is. Placeholder will get replaced with '75%'.")
+    }
+    titleLabel.text = String(format: format, formatter.string(from: NSNumber(value: friendlyRatio))!)
+    
+    // Legend
+    friendlyLegendLabel.text = Loc.FriendlyPath
+    unfriendlyLegendLabel.text = Loc.UnfriendlyPath
+    unknownLegendLabel.text = Loc.UnknownPathFriendliness
+    
+    // Update bar chart.
+    let widthConstraints = [
+        friendlyBarView.widthAnchor.constraint(equalTo: widthAnchor, multiplier: CGFloat(friendlyRatio)),
+        unfriendlyBarView.widthAnchor.constraint(equalTo: widthAnchor, multiplier: CGFloat(unfriendlyRatio)),
+        unknownBarView.widthAnchor.constraint(equalTo: widthAnchor, multiplier: CGFloat(unknownRatio))
+      ]
+    
+    // Lower the priority of the width constraint because floating point arithmetic may produce
+    // ratios that don't add up to precisely 1 should one of the bar views has zero width.
+    widthConstraints.forEach { $0.priority = 999 }
+    NSLayoutConstraint.activate(widthConstraints)
+    
+    let distanceFormatter = MKDistanceFormatter()
```

Comment:
Can you add the significant digits (two?)  there? Our data isn't precise on a metre level, so the UI shouldn't pretend it is.

Reply:
`MKDistanceFormatter` doesn't allow specifying significant digits though. `NumberFormatter` does, but `MKDistanceFormatter` isn't a subclass. `MKDistanceFormatter` also rounds the results and chooses metre/kilometre itself (cf. `NumberFormatter`). It rounds 161m to 160m, 1463m to 1.5km (also localised)

Reply:
Copy that. All good then.